src/components/GoogleAuth.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { googleSignIn, googleSignOut } from '../actions';
+import { GOOGLE_AUTH_PUBLIC } from '../apis/constants';
+
+class GoogleAuth extends React.Component {
+    componentDidMount() {
+        window.gapi.load('client:auth2', () => {
+          window.gapi.client
+            .init({
+              clientId: GOOGLE_AUTH_PUBLIC,
+              scope: 'email'
+            })
+            .then(() => {
+              this.auth = window.gapi.auth2.getAuthInstance();
```

Comment:
I'm not a big fan of assigning values to `this` directly. Did you see this in documentation somewhere that says this is the preferred approach?

Reply:
I assigned it to `this` for prototyping as I wasn't sure where to store the AuthInstance. Where do you think that should get stored in?